### DISCUSSION TITLE
How to distinguish two zod errors?

### DISCUSSION THREAD

**Discussion by yanisurbis** [colinhacks/zod#2337] [Q&A]
I have multiple layers in my application and I want to handle zod errors from different layers differently. I was looking for some way to add meta information to schemas (like https://clojure.org/reference/metadata#_with_meta_obj_map) but it looks like there is no way.
What would you suggest? Thank you.

**Comment by JacobWeisenburger** [comment]
what type of meta information are you wanting to add? Strings? Plain Objects? Complex Objects? Something else?

**Comment by yanisurbis** [comment]
@JacobWeisenburger thank you for your reply, simple string would be enough
